compiler/rustc_feature/src/builtin_attrs.rs

```diff
@@ -752,6 +752,11 @@ pub const BUILTIN_ATTRIBUTES: &[BuiltinAttribute] = &[
         template!(NameValueStr: "transparent|semitransparent|opaque"), ErrorFollowing,
         EncodeCrossCrate::Yes, "used internally for testing macro hygiene",
     ),
+    rustc_attr!(
+        rustc_autodiff, Normal,
+        template!(Word, List: r#""...""#), DuplicatesOk,
```

Comment:
Problem: `DuplicatesOk` seems a bit sus to me

```rs
#[autodiff(df, ...)]
#[autodiff(df, ...)]
fn foo() {}
```

This seems sus? I suspect we actually want `ErrorFollowing` or `ErrorPreceding` depending on which attribute order we want to actually honor.

Reply:
It's not a bug, it's a feature! And a cool one on top, which e.g. JAX doesn't support to this extend.

If you have two numerical solvers that use derivatives of different arguments for their optimizations, you can just generate df_dx and df_dy as two different functions with different activities for autodiff.

There aren't much doc comments around here but I have it described in the docs, so I would leave it at that?

Or if you have vector based functions and don't know at compile time how large input and output dimensions will be, then you can create df_fwd and df_rev and decide at runtime which of the two to call to get the best performance.

Those